libs/SmartSync/src/com/salesforce/androidsdk/smartsync/util/SoqlSyncDownTarget.java

```diff
@@ -130,6 +138,80 @@ public JSONArray continueFetch(SyncManager syncManager) throws IOException, JSON
         return records;
     }
 
+    @Override
+    public Set<String> getListOfRemoteIds(SyncManager syncManager, Set<String> localIds) {
```

Comment:
Makes sense. I'll switch it to run with a modified ```SELECT Id``` query - should be fairly straightforward, except for the paging bit.